build-gradle/build.sh

```diff
@@ -0,0 +1,238 @@
+    args+=("-Dsonar.analysis.repository=$GITHUB_REPOSITORY")
+  fi
+
+  # Artifactory deployment
+  if should_deploy; then
+    args+=("artifactoryPublish")
+  fi
+
+  # Build number
+  args+=("-DbuildNumber=$BUILD_NUMBER")
+
+  # Additional arguments
+  if [[ -n "$GRADLE_ARGS" ]]; then
+    read -ra extra_args <<< "$GRADLE_ARGS"
+    args+=("${extra_args[@]}")
+  fi
+
+  echo "${args[@]}"
+}
+
+should_deploy() {
+  local pull_request="${PULL_REQUEST:-false}"
+
+  if [[ "$pull_request" != "false" ]]; then
+    # For pull requests, deploy only if explicitly enabled
+    [[ "$DEPLOY_PULL_REQUEST" == "true" ]]
+  else
+    [[ "$GITHUB_REF_NAME" == "master" ]] || \
+    [[ "$GITHUB_REF_NAME" == branch-* ]] || \
+    [[ "$GITHUB_REF_NAME" == dogfood-on-* ]] || \
+    [[ "$GITHUB_REF_NAME" == feature/long/* ]]
+  fi
+}
+
+set_sonar_args() {
+  local -n args_ref=$1
+  local pull_request="${PULL_REQUEST:-false}"
+
+  if [[ -z "${SONAR_HOST_URL:-}" || -z "${SONAR_TOKEN:-}" ]]; then
+    return 0
+  fi
+
+  if [[ "$GITHUB_REF_NAME" == "master" && "$pull_request" == "false" ]]; then
+    # Master branch analysis
+    args_ref+=("-Dsonar.projectVersion=$INITIAL_VERSION")
+    args_ref+=("-Dsonar.analysis.sha1=$GITHUB_SHA")
+
+  elif [[ "$GITHUB_REF_NAME" == branch-* && "$pull_request" == "false" ]]; then
+    # Maintenance branch analysis
+    args_ref+=("-Dsonar.branch.name=$GITHUB_REF_NAME")
+    args_ref+=("-Dsonar.projectVersion=$INITIAL_VERSION")
+    args_ref+=("-Dsonar.analysis.sha1=$GITHUB_SHA")
+
+  elif [[ "$pull_request" != "false" ]]; then
+    # Pull request analysis
+    args_ref+=("-Dsonar.analysis.sha1=$PULL_REQUEST_SHA")
+    args_ref+=("-Dsonar.analysis.prNumber=$PULL_REQUEST")
+
+  elif [[ "$GITHUB_REF_NAME" == feature/long/* && "$pull_request" == "false" ]]; then
+    # Long-lived feature branch analysis
+    args_ref+=("-Dsonar.branch.name=$GITHUB_REF_NAME")
+    args_ref+=("-Dsonar.analysis.sha1=$GITHUB_SHA")
+  fi
+}
+
+gradle_build() {
+  # Setup Gradle
+  if command_exists gradle; then
+    GRADLE_CMD="gradle"
+  elif [[ -f "./gradlew" ]]; then
+    GRADLE_CMD="./gradlew"
+    chmod +x ./gradlew
+  else
+    echo "Neither gradle nor gradlew found" >&2
+    return 1
+  fi
+
+  echo "Using Gradle command: $GRADLE_CMD"
+  export GRADLE_CMD
+
+  # Conditional validation for SonarQube
+  if [[ -n "${SONAR_HOST_URL:-}" && -z "${SONAR_TOKEN:-}" ]]; then
+    echo "SONAR_TOKEN is required when SONAR_HOST_URL is set" >&2
+    return 1
+  fi
+
+  # Conditional validation for Artifactory
+  if [[ -n "${ARTIFACTORY_DEPLOY_REPO:-}" && ( -z "${ARTIFACTORY_DEPLOY_USERNAME:-}" || -z "${ARTIFACTORY_DEPLOY_PASSWORD:-}" ) ]]; then
+    echo "ARTIFACTORY_DEPLOY_USERNAME and ARTIFACTORY_DEPLOY_PASSWORD are required when ARTIFACTORY_DEPLOY_REPO is set" >&2
+    return 1
+  fi
+
+  local gradle_args
+  read -ra gradle_args <<< "$(build_gradle_args)"
+
+  set_sonar_args gradle_args
+
+  local build_type
+  local pull_request="${PULL_REQUEST:-false}"
```

Comment:
This is duplicated everywhere.
All you need is 
```
is_pull_request() {
  [[ "$GITHUB_EVENT_NAME" == "pull_request" ]]
}
```

and then `if is_pull_request`

Or use directly `[[ "$GITHUB_EVENT_NAME" == "pull_request" ]]`

This `"$pull_request" == "false"` is a debt in the original script. Adding complexity for no reason.